### PR TITLE
types(jasminewd2): add new typings for jasminewd2

### PR DESCRIPTION
**Pull request by cnishina**
- [X] Make your PR against the `master` branch.
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [X] Run `tsc` without errors.

If adding a new definition:
- [X] The package does not provide its own types, and you can not add them.
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis` set to `true`.

### CODE REVIEW

File: jasminewd/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
+// Type definitions for jasminewd2 0.1.1
+// Project: https://github.com/angular/jasminewd
+// Definitions by: Sammy Jelin <https://github.com/sjelin>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+//
+// This is jasminewd's internal version of @types/jasminewd2.  If you need types
```
Comment (sjelin) [inline jasminewd/index.d.ts]: no it isn't :smile:

**Review by sjelin — CHANGES_REQUESTED** [review]

File: jasminewd/index.d.ts (hunk 1 above)
Comment (cnishina) [inline jasminewd/index.d.ts]: Removed comment.

File: jasminewd/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
+// Type definitions for jasminewd2 0.1.1
```
Comment (mhegazy) [inline jasminewd/index.d.ts]: please add a [tslint.json](https://github.com/DefinitelyTyped/DefinitelyTyped#lint), it should catch issues like this header version being invalid.

**Review by mhegazy — CHANGES_REQUESTED** [review]

File: jasminewd/index.d.ts (hunk 2 above)
Comment (cnishina) [inline jasminewd/index.d.ts]: Moving the folder to jasminewd2. The published npm is "jasminewd2" however, the project is called "jasminewd"

**Comment by cnishina** [comment]
Removed tslint check:

```
  interface AsyncCustomMatcherFactory {
      (util: MatchersUtil, customEqualityTesters: CustomEqualityTester[]): AsyncCustomMatcher;
  }
```

This causes the tslint to be unhappy. It recommends changing this to a function.

@sjelin for review and comment.

**Review by sjelin — CHANGES_REQUESTED** [review]
test `addMatchers` and it should be good

**Comment by mhegazy** [comment]
> Removed tslint check:

do not remove the tslint.json, just disable the rule in the config file.

**Comment by cnishina** [comment]
Hmmm... I'm not sure how to do this. Could you point me to the right documentation?

**Comment by cnishina** [comment]
@mhegazy I added tslint.json back. I'm not sure how to do disable the tslint from the config (I'm guessing this is done in tsconfig). Could you point me to the right documentation?

**Comment by mhegazy** [comment]
change your tslint file to:

```ts
{
    "extends": "../tslint.json",
    "rules": {
        "callable-types": false
    }
}
```

File: jasminewd2/jasminewd2-tests.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+              };
+            }
+          };
+        },
+        // Example custom matcher returning a promise that resolves to true/false.
+        toBeDisplayed: function() {
+          return {
+            compare: function(actual: any, expected: void) {
+              return {
+                pass: actual.isDisplayed()
```
Comment (sjelin) [inline jasminewd2/jasminewd2-tests.ts]: I'd prefer you put a `Promise<boolean>` here to make the test more explicit

File: jasminewd2/jasminewd2-tests.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+              };
+            }
+          };
+        }
+      };
+      jasmine.addMatchers(matchers);
```
Comment (sjelin) [inline jasminewd2/jasminewd2-tests.ts]: For completeness, add:
```ts
jasmine.getEnv().addMatchers(matchers);
jasmine.getEnv().currentSpec.addMatchers(matchers);
```

**Review by sjelin — CHANGES_REQUESTED** [review]
Minor stuff

File: jasminewd2/jasminewd2-tests.ts (hunk 3 above)
Comment (cnishina) [inline jasminewd2/jasminewd2-tests.ts]: Done.

File: jasminewd2/jasminewd2-tests.ts (hunk 4 above)
Comment (cnishina) [inline jasminewd2/jasminewd2-tests.ts]: Done.

**Comment by cnishina** [comment]
@mhegazy Updated the tslint. Please review.

**Review by mhegazy — APPROVED** [review]

**Pull request merged**